Pod/Classes/VOKCoreDataManager.m

```diff
@@ -484,78 +495,95 @@ - (NSManagedObjectContext *)safeContext:(NSManagedObjectContext *)context
 
 - (void)saveMainContext
 {
-    if ([NSOperationQueue mainQueue] == [NSOperationQueue currentQueue]) {
-        [self saveContext:self.managedObjectContext];
-    } else {
-        [self.managedObjectContext performBlock:^{
-            [self saveContext:self.managedObjectContext];
-        }];
-    }
+    [self saveContext:self.managedObjectContext andWait:NO];
 }
 
 - (void)saveMainContextAndWait
 {
-    if ([NSOperationQueue mainQueue] == [NSOperationQueue currentQueue]) {
-        [self saveContext:self.managedObjectContext];
-    } else {
-        [self.managedObjectContext performBlockAndWait:^{
-            [self saveContext:self.managedObjectContext];
-        }];
-    }
+    [self saveContext:self.managedObjectContext andWait:YES];
 }
 
-- (void)saveContext:(NSManagedObjectContext *)context
+- (void)obtainPermanentIDsForInsertionsInContext:(NSManagedObjectContext *)context
 {
-    NSError *error;
-    if (![context save:&error]) {
-        VOK_CDLog(@"Unresolved error %@, %@", error, [error localizedDescription]);
-    }
+    [context performBlockAndWait:^{
+        NSError *error;
+        if (![context obtainPermanentIDsForObjects:context.insertedObjects.allObjects
+                                             error:&error]) {
+            VOK_CDLog(@"Error obtaining permanent ID for insertions: %@", error);
+        }
+    }];
 }
 
-- (void)saveTempContext:(NSManagedObjectContext *)tempContext
+/**
+ Save any changes to a managed object context to the persistent store.
+ If a context has no changes, no saves will be performed.
+ 
+ @param context The context to save
+ @param wait Whether to save synchronously (YES) or asynchronously (NO)
+ */
+- (void)saveContext:(NSManagedObjectContext *)context
+            andWait:(BOOL)wait
 {
-    [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(tempContextSaved:)
-                                                 name:NSManagedObjectContextDidSaveNotification
-                                               object:tempContext];
+    __block BOOL hasChanges;
+    [context performBlockAndWait:^{
```

Comment:
explain why `performBlockAndWait` instead of `performBlock` or explain why it's done using the block method at all?